packages/frontend/core/src/modules/share-doc/stores/share-reader.ts

```diff
@@ -0,0 +1,42 @@
+import { ErrorNames, UserFriendlyError } from '@affine/graphql';
+import { type DocMode, Store } from '@toeverything/infra';
+
+import { type FetchService, isBackendError } from '../../cloud';
+
+export class ShareReaderStore extends Store {
+  constructor(private readonly fetchService: FetchService) {
+    super();
+  }
+
+  async loadShare(workspaceId: string, docId: string) {
+    try {
+      const docResponse = await this.fetchService.fetch(
+        `/api/workspaces/${workspaceId}/docs/${docId}`
+      );
+      const publishMode = docResponse.headers.get(
+        'publish-mode'
+      ) as DocMode | null;
+      const docBinary = await docResponse.arrayBuffer();
+
+      const workspaceResponse = await this.fetchService.fetch(
+        `/api/workspaces/${workspaceId}/docs/${workspaceId}`
+      );
+      const workspaceBinary = await workspaceResponse.arrayBuffer();
```

Comment:
I think this is a security breach that leaks metadata of the whole workspace even when the user has no access to the workspace.
Is that necessary to load it for shared page?

Reply:
track in AF-1259